documentation/sphinx/source/developer-guide.rst

```diff
@@ -586,6 +586,8 @@ Conflicts can be *avoided*, reducing isolation, in two ways:
 
 Conflicts can be *created*, increasing isolation, by :ref:`explicitly adding <api-python-conflict-ranges>` read or write conflict ranges. 
 
+.. warning:: Ordering of *write* operations and *add read conflict range* operations should be considered carefully as it could result in unexpected behavior. Adding read conflict ranges will skip adding conflicts for the range of keys that have been written previously in the transaction. Adding them before the respective writes will add to the conflict ranges.
```

Comment:
```suggestion
.. note:: *add read conflict range* behaves as if the client is reading the range. This means *add read conflict range* will not add conflict ranges for keys that have been written earlier in the same transaction. This is the intended behavior, as it allows users to compose transactions together without introducing unnecessary conflicts.

Ordering of *write* operations and *add read conflict range* operations should be considered carefully as it could result in unexpected behavior. Adding read conflict ranges will skip adding conflicts for the range of keys that have been written previously in the transaction. Adding them before the respective writes will add to the conflict ranges.
```